src/components/navbar.js

```diff
@@ -50,7 +50,7 @@ const Navbar = (prop) => {
     window.addEventListener('scroll', scrollListener)
 
     return () => window.removeEventListener('scroll', scrollListener)
-  }, [])
+  }, [locale, location.pathname])
```

Comment:
I think the logic of the `scroll part` should be extracted into a new useEffect.